src/libaktualizr/primary/aktualizr_test.cc

```diff
@@ -677,23 +677,25 @@ class HttpFakeNoCorrelationId : public HttpFake {
 TEST(Aktualizr, FullNoCorrelationId) {
   TemporaryDirectory temp_dir;
   auto http = std::make_shared<HttpFakeNoCorrelationId>(temp_dir.Path());
-  Config conf = makeTestConfig(temp_dir, http->tls_server);
+  {
+    Config conf = makeTestConfig(temp_dir, http->tls_server);
 
-  auto storage = INvStorage::newStorage(conf.storage);
-  auto sig = std::make_shared<boost::signals2::signal<void(std::shared_ptr<event::BaseEvent>)>>();
-  auto up = SotaUptaneClient::newTestClient(conf, storage, http, sig);
-  Aktualizr aktualizr(conf, storage, up, sig);
+    auto storage = INvStorage::newStorage(conf.storage);
+    auto sig = std::make_shared<boost::signals2::signal<void(std::shared_ptr<event::BaseEvent>)>>();
+    auto up = SotaUptaneClient::newTestClient(conf, storage, http, sig);
+    Aktualizr aktualizr(conf, storage, up, sig);
 
-  aktualizr.Initialize();
-  UpdateCheckResult update_result = aktualizr.CheckUpdates();
-  EXPECT_EQ(update_result.status, UpdateStatus::kUpdatesAvailable);
+    aktualizr.Initialize();
+    UpdateCheckResult update_result = aktualizr.CheckUpdates();
+    EXPECT_EQ(update_result.status, UpdateStatus::kUpdatesAvailable);
 
-  DownloadResult download_result = aktualizr.Download(update_result.updates);
-  EXPECT_EQ(download_result.status, DownloadStatus::kSuccess);
+    DownloadResult download_result = aktualizr.Download(update_result.updates);
+    EXPECT_EQ(download_result.status, DownloadStatus::kSuccess);
 
-  InstallResult install_result = aktualizr.Install(download_result.updates);
-  for (const auto& r : install_result.reports) {
-    EXPECT_EQ(r.status.result_code, data::UpdateResultCode::kOk);
+    InstallResult install_result = aktualizr.Install(download_result.updates);
+    for (const auto& r : install_result.reports) {
+      EXPECT_EQ(r.status.result_code, data::UpdateResultCode::kOk);
+    }
```

Comment:
Not that this is bad or something, but why the extra scope enclosure?

Reply:
It's explained in the commit message. I've seen two spurious failures of this test and I think that's because we count the number of events before the queue had the time to be flushed.

Reply:
That said, it probably needs a comment, good point :)

Reply:
Oh, perfect, now I get it. Sorry, I do normally read the commits but forgot this time.